### ISSUE TITLE
S1125 FP with nullable booleans in ternary expressions C#9

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#5219]
### Description

C#9 allows to have target typed ternary expressions if both branches can be converted to the target type. In this case the rule raises a false positive.

### Repro steps

```
bool B { get; set; }

void M()
{
    bool? v = B ? true : null; // S1125 on this line
}
```

### Expected behavior

No issue is raised.

### Actual behavior

An issue is raised.

### Related information

* C#/VB.NET Plugins version: 8.33.0.40503
